### PR TITLE
More user friendly errors in layout.js

### LINKED ISSUE #26398: <amp-script> outputs error to console for layout != 'container'
Whenever I use <amp-script> with a layout that's not `container` - e.g., `fixed` or `responsive`, the following error is output to the console:

`Expected height to be available:  null` - `error.js:195`

While this may seem trivial, I've already seen examples where AMP developers reached out to Googlers for help because they couldn't figure out how to debug their code, and they assumed this bug was their fault.

See screenshot for full stacktrace:

Screen Shot 2020-01-17 at 7 02 46 PM

### PR DESCRIPTION
**Pull request by choumx**
Fixes #26398.

### CODE REVIEW

File: src/layout.js [hunk 1]
```diff
@@ -375,15 +375,23 @@ export function applyStaticLayout(element) {
 
   // Input layout attributes.
   const inputLayout = layoutAttr ? parseLayout(layoutAttr) : null;
-  userAssert(inputLayout !== undefined, 'Unknown layout: %s', layoutAttr);
+  userAssert(
+    inputLayout !== undefined,
+    'Invalid "layout" value: %s',
```
Comment (morsssss) [inline src/layout.js]: Is it relatively trivial to include in this error message the name of the offending component?

File: src/layout.js [hunk 2]
```diff
@@ -375,15 +375,23 @@ export function applyStaticLayout(element) {
...
+  );
   /** @const {string|null|undefined} */
   const inputWidth =
     widthAttr && widthAttr != 'auto' ? parseLength(widthAttr) : widthAttr;
-  userAssert(inputWidth !== undefined, 'Invalid width value: %s', widthAttr);
+  userAssert(inputWidth !== undefined, 'Invalid "width" value: %s', widthAttr);
   /** @const {string|null|undefined} */
   const inputHeight =
     heightAttr && heightAttr != 'fluid' ? parseLength(heightAttr) : heightAttr;
-  userAssert(inputHeight !== undefined, 'Invalid height value: %s', heightAttr);
```
Comment (samouri) [inline src/layout.js]: For my own learning, why do you prefer variable interpolation via `%s` instead of via backticks?

Comment (choumx) [reply src/layout.js]: We have a feature that extracts error strings at compile time and replaces them with an error code for a smaller binary size. It doesn't play nice with JS string templating unfortunately.

Comment (choumx) [reply src/layout.js]: Actually I'm poking around now and I'm not sure this is true and/or if the feature was launched. :)

File: src/layout.js (hunk 1 above)
Comment (choumx) [reply src/layout.js]: Added the element itself. 👍

**Comment by choumx** [comment]
Any other comments or can we merge? :)

**Comment by morsssss** [comment]
@choumx I admit I did make another suggestion about *even friendlier* error messages (is such a thing even possible?) Please see above....

**Comment by choumx** [comment]
@morsssss Are you talking about https://github.com/ampproject/amphtml/pull/26448#discussion_r369863366? I made a change for that already.

**Comment by morsssss** [comment]
@choumx I don't know if I can get a direct link to this particular comment - but it's the one above that suggests language like

`<amp-foo>: Invalid "width" value: 300`

and 

`<amp-foo>: length units should be the same for "width" and "height"`

instead of like 

`Length units should be the same for "width" and "height": px, em, <amp-foo>`

**Comment by choumx** [comment]
As is, devtools will render the element itself into console (e.g. supports highlight on hover). Try it out and let me know if you'd still prefer a prefix tag string.

**Comment by morsssss** [comment]
That highlight on hover will be excellent! (I don't get this yet in my v0.js) But I think that if it's just as easy to output friendlier, more human-readable errors, we should. I think that, otherwise, we're making ourselves less accessible than other programming frameworks.

That said, this is not matter of broken functionality, so I'm not going to belabor the point :)

**Comment by morsssss** [comment]
just wanted to bump this one up again...

**Comment by choumx** [comment]
Here's what the error looks like now: 
image

I think that's what you wanted right?

**Comment by morsssss** [comment]
That's really nice!

**Pull request merged**